src/vpp_plugins/vpp-chaperone/VppConfigManager.cpp

```diff
@@ -1746,23 +1754,38 @@ void VppConfigManager::doCpePolicerConfig (VppClient &vppClient,
           policerConfig.isAdd = 0; // Delete
           vppClient.addDelPolicer (policerConfig, policerIndex);
         }
-      policerConfig.isAdd = 1; // Add
-      vppClient.addDelPolicer (policerConfig, policerIndex);
-
-      // Create session matching to AFx1: upstream traffic is expected to be
-      // marked with DSCP corresponding to a traffic class with low drop
-      // preference (green)
-      VppClient::ClassifierSessionConfig sessionConfig;
-      std::vector<u8> matchVector = {
-          0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, kAFxyMatch.at (tc), 0x80};
-      sessionConfig.match = matchVector.data ();
-      sessionConfig.hitNextIndex = policerIndex;
-      sessionConfig.tableIndex = tableConfig.tableIndex;
-      vppClient.addDelClassifierSession (sessionConfig);
-    }
-  // Map table to interface
-  vppClient.setClassifierTableNetif (true, interfaceIndex,
-                                     tableConfig.tableIndex);
+      if (policerPresent)
+        {
+          policerConfig.isAdd = 1; // Add
+          vppClient.addDelPolicer (policerConfig, policerIndex);
+        }
+      else if (!policerPresent)
+        {
+          policerConfig.isAdd = 0; // Delete
+          vppClient.addDelPolicer (policerConfig, policerIndex);
+          continue;
+        }
+      if (policerPresent)
+        {
+          // Create session matching to AFx1: upstream traffic is expected to
+          // be marked with DSCP corresponding to a traffic class with low drop
+          // preference (green)
+          VppClient::ClassifierSessionConfig sessionConfig;
+          std::vector<u8> matchVector = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+              0, 0, kAFxyMatch.at (tc), 0x80};
+          sessionConfig.match = matchVector.data ();
+          sessionConfig.hitNextIndex = policerIndex;
+          sessionConfig.tableIndex = tableConfig.tableIndex;
+          vppClient.addDelClassifierSession (sessionConfig);
+        }
```

Comment:
This if-elseif-if chain doesn't make sense. Write it like this:
```cpp
      if (policerPresent)
        {
          policerConfig.isAdd = 1; // Add
          vppClient.addDelPolicer (policerConfig, policerIndex);

          // Create session matching to AFx1: upstream traffic is expected to
          // be marked with DSCP corresponding to a traffic class with low drop
          // preference (green)
          VppClient::ClassifierSessionConfig sessionConfig;
          std::vector<u8> matchVector = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
              0, 0, kAFxyMatch.at (tc), 0x80};
          sessionConfig.match = matchVector.data ();
          sessionConfig.hitNextIndex = policerIndex;
          sessionConfig.tableIndex = tableConfig.tableIndex;
          vppClient.addDelClassifierSession (sessionConfig);
        }
      else
        {
          policerConfig.isAdd = 0; // Delete
          vppClient.addDelPolicer (policerConfig, policerIndex);
        }
```

Reply:
Addressed Review Comment.